Userland/Libraries/LibCrypto/Cipher/AES.cpp

```diff
@@ -152,24 +147,259 @@ void AESCipherKey::expand_encrypt_key(ReadonlyBytes user_key, size_t bits)
     }
 }
 
-void AESCipherKey::expand_decrypt_key(ReadonlyBytes user_key, size_t bits)
+#if (ARCH(I386) || ARCH(X86_64))
+static AK::SIMD::i64x2 AESCipherKey_expand_encrypt_key_128_assist(AK::SIMD::i64x2 const& a, AK::SIMD::i64x2 const& b)
+{
+    AK::SIMD::i32x4 ta {};
+    AK::SIMD::i32x4 tb {};
+    AK::SIMD::i32x4 tc {};
+
+    ta = bit_cast<AK::SIMD::i32x4>(a);
+    tb = bit_cast<AK::SIMD::i32x4>(b);
+    tb = AK::SIMD::i32x4 { tb[3], tb[3], tb[3], tb[3] };
+    tc = AK::SIMD::i32x4 { 0, ta[0], ta[1], ta[2] };
+    ta ^= tc;
+    tc = AK::SIMD::i32x4 { 0, ta[0], ta[1], ta[2] };
+    ta ^= tc;
+    tc = AK::SIMD::i32x4 { 0, ta[0], ta[1], ta[2] };
+    ta ^= tc;
+    ta ^= tb;
+    return bit_cast<AK::SIMD::i64x2>(ta);
+}
+
+static void __attribute__((target("aes"), used)) AESCipherKey_expand_encrypt_key_128(u32* round_key, ReadonlyBytes user_key)
+{
+    auto const* user_ptr = user_key.data();
+    AK::SIMD::i64x2 ta {};
+    AK::SIMD::i64x2 tb {};
+
+    ta = AK::SIMD::load_unaligned<AK::SIMD::i64x2>(&user_ptr[0 * 16]);
+    AK::SIMD::store_unaligned(&round_key[0 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x01);
+    ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb);
+    AK::SIMD::store_unaligned(&round_key[1 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x02);
+    ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb);
+    AK::SIMD::store_unaligned(&round_key[2 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x04);
+    ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb);
+    AK::SIMD::store_unaligned(&round_key[3 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x08);
+    ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb);
+    AK::SIMD::store_unaligned(&round_key[4 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x10);
+    ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb);
+    AK::SIMD::store_unaligned(&round_key[5 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x20);
+    ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb);
+    AK::SIMD::store_unaligned(&round_key[6 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x40);
+    ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb);
+    AK::SIMD::store_unaligned(&round_key[7 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x80);
+    ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb);
+    AK::SIMD::store_unaligned(&round_key[8 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x1b);
+    ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb);
+    AK::SIMD::store_unaligned(&round_key[9 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x36);
```

Comment:
Can this be done less repetitively?
or at least split the iterations by a newline

same goes for the algorithms below

Reply:
OK, visually sepratated.